### PR TITLE
feat(docs): add support for multiple documentation instances

### LINKED ISSUE #94: Support multiple docs instances/pages with configurable folders and route mounting
## Overview

Implement support for multiple documentation instances/pages in the project, following the roadmap direction. This feature should allow users to:

- Configure the folder(s) where docs are located
- (Potentially) configure the loader responsible for docs
- Specify the path(s) under which their routes should be mounted

## Implementation notes
- Investigate the best approach for Astro to support multiple docs roots. Consider whether handling via loader config or direct folder selection is better for flexibility and scalability.
- Leverage the new parameter in Astro's `getStaticPaths` function, particularly the `routePattern`, to enable mounting docs routes under user-defined paths. Reference: [Astro Routing Documentation](https://docs.astro.build/en/reference/routing-reference/#routepattern)

## Tasks
- [ ] Analyze Astro docs and examples for multi-doc setup
- [ ] Provide config options to set docs folders/roots
- [ ] Allow mounting of docs at custom route paths via `routePattern`
- [ ] Document how integrators can use this feature
- [ ] (Optional) Support custom loader implementation for advanced use cases

## References
- [Astro Routing Documentation](https://docs.astro.build/en/reference/routing-reference/#routepattern)
- Internal roadmap

@claude 
---

This issue is based on the roadmap and current direction for handling dynamic or multiple documentation spaces within the project.

### PR DESCRIPTION
**Pull request by levino**
## Summary
- Add `DocsConfig` interface with `routeBasePath` option for custom route mounting
- Add `createDocsCollection()` helper for defining docs content collections
- Add route utilities (`getDocPath`, `getRouteParams`) for custom routes
- Update `DocsLayout` to accept configuration props
- Add comprehensive documentation for multi-docs feature

Closes #94

Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

**Comment by levino** [comment]
@claude. Please add to your claude.md that you always need to

- update the documentation
- add a changeset (for each changed package) and
- all new features need extensive testing via playwright and potentially additionally some unit tests with vite.

Write it down for your future self.

Then add tests, examples in the demo apps (or create new ones, maybe, if it makes sense), update the docs and add a changeset if any of these is not yet implemented.

File: CLAUDE.md [hunk 1]
```diff
@@ -155,7 +155,15 @@ Documentation features:
...
-   - Include a clear description of what changed and why
+   - **Write user-centric descriptions**: Focus on what users can now do, not what code was changed
+     - ✅ Good: "You can now host multiple documentation sections with custom URL paths"
+     - ❌ Bad: "Added `DocsConfig` interface with `routeBasePath` option"
+   - Explain the benefit and use case, not the implementation details
+9. **Testing Requirements**: All new features require extensive testing:
+   - Add Playwright E2E tests for user-facing features (in `apps/demos/*/tests/e2e/`)
+   - Add unit tests with Vitest when appropriate for utility functions and business logic
+   - Tests should cover both happy paths and edge cases
+   - Run `npm run test:e2e` to verify all tests pass before submitting changes
```
Comment (levino) [inline CLAUDE.md]: @claude very good. also add that all demos have a "docusaurus counterpart" which has the same content and is used to compare the outcome of docusaurus and shipyard for the very same data. So it is important to always keep the docusaurus demos in sync with our demos.
